docs/language/api.rst

```diff
@@ -179,6 +179,59 @@ other case, the first false value will be returned. Example usage:
     False
 
 
+as->
+----
+
+.. versionadded:: 0.12.0
+
+Expands to sequence of assignments to the provided name, starting with head.
+The previous result is thus available in the subsequent form. Returns the final
+result, and leaves the name bound to it in the local scope. This behaves much
+like the other threading macros, but requires you to specify the threading
+point per form via the name instead of always the first or last arument.
```

Comment:
Instead of ``assigned-name`` I could also us ``it`` as it's short and generic enough. I'll try to come up with a better examples still, that would highlight usage of ``as->`` better (as now the example could be easily rewritten with ``->``).